classes-es6/index.html

```diff
@@ -0,0 +1,246 @@
+        <li>Extending an existing class</li>
+        <li>Subclassing methods of a parent class</li>
+        <li>Defining static methods</li>
+        <li>Subclassing built-ins (not yet supported)</li>
+       </ul>
+    </p>
+
+    <!-- // [START code-block] -->
+    <div class="output">
+        <pre id="log"></pre>
+    </div>
+
+    <script>
+        "use strict";
+
+        function log() {
+            document.querySelector('#log').textContent += Array.prototype.join.call(arguments, '') + '\n';
+        }
+
+        // Example 1: Creating a new class (declaration-form)
+        // ===============================================================
+
+        // A base class is defined using the new reserved 'class' keyword
+        class Polygon {
+            // ..and an (optional) custom class constructor. If one is
+            // not supplied, a default constructor is used instead:
+            // constructor() { }
+            constructor(height, width) {
+                this.name = 'Polygon';
+                this.height = height;
+                this.width = width;
+            }
+
+            // Simple class instance methods using short-hand method
+            // declaration
+            sayName() {
+                log('Hi, I am a ', this.name + '.');
+            }
+
+            sayHistory() {
+                log('"Polygon" is derived from the Greek polus (many) and gonia (angle).')
+            }
+
+            // We will look at static and subclassed methods shortly
+        }
+
+        // Classes are used just like ES5 constructor functions:
+        let p = new Polygon(300,400);
+        p.sayName();
+        log('The width of this polygon is ' + p.width);
+
+
+        // Example 2: Creating a new class (expression-form)
+        // ===============================================================
+
+        // Our Polygon class above is an example of a Class declaration.
+        // ES6 classes also support Class expressions - just another way
+        // of defining a new class. For example:
+        const MyPoly = class Poly {
+            getPolyName() {
+                log('Hi. I was created with a Class expression. My name is ' + Poly.name);
+            }
+        }
+
+        let inst = new MyPoly();
+        inst.getPolyName();
+
+
+        // Example 3: Extending an existing class
+        // ===============================================================
+
+        // Classes support extending other classes, but can also extend
+        // other objects. Let's extend the Polygon class to create a new
+        // derived class called Square.
+        class Square extends Polygon {
+            constructor(length) {
+                // The reserved 'super' keyword allows access to parent methods.
+                // Here, it will call the parent class' constructor with lengths
+                // provided for the Polygon's width and height
+                super(length, length);
```

Comment:
Note: `super()` for making super-constructor calls and `super.prop` for accessing super-properties and making super-method calls are quite different. But maybe that’s too subtle to be explained here.

Possibly: Mention what the default constructor for derived classes looks like.